# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: b55aacf910e7
# Full Hash: b55aacf910e704508a5820c09b25411142ccf7cb
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-25 09:36:02
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 4: Make ContentParent KeepAlives explicit with RAII references, r=smaug,dom-worker-reviewers,asuth
#   
#   This is a fairly significant patch, however it would be difficult to break it
#   down into smaller patches:
#   
# ==============================================================================

diff -r 3dd07ed49638 -r b55aacf910e7 dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Mon Jun 24 23:19:28 2024 +0000
+++ b/dom/ipc/BrowserParent.h	Mon Jun 24 23:19:28 2024 +0000
@@ -19,6 +19,7 @@
 #include "mozilla/dom/BrowserBridgeParent.h"
 #include "mozilla/dom/PBrowserParent.h"
 #include "mozilla/dom/TabContext.h"
+#include "mozilla/dom/UniqueContentParentKeepAlive.h"
 #include "mozilla/dom/VsyncParent.h"
 #include "mozilla/dom/ipc/IdType.h"
 #include "mozilla/layout/RemoteLayerTreeOwner.h"
@@ -872,6 +873,12 @@
   // non-null.
   BrowserHost* mBrowserHost;
 
+  // KeepAlive for the containing process.
+  // NOTE: While this is a strong reference to ContentParent, which is
+  // cycle-collected, it is cleared as the BrowserParent's IPC connection is
+  // destroyed, so does not need to be cycle-collected.
+  UniqueContentParentKeepAlive mContentParentKeepAlive;
+
   ContentCacheInParent mContentCache;
 
   layout::RemoteLayerTreeOwner mRemoteLayerTreeOwner;